x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_enrichment/types.ts

```diff
@@ -6,13 +6,26 @@
  */
 
 import {
-  DissectProcessingDefinition,
+  Condition,
+  DissectProcessorConfig,
   FieldDefinitionConfig,
-  GrokProcessingDefinition,
-  ProcessingDefinition,
+  GrokProcessorConfig,
 } from '@kbn/streams-schema';
 
-export interface ProcessorDefinition extends ProcessingDefinition {
+export interface DissectProcessingDefinition {
+  dissect: Omit<DissectProcessorConfig, 'if'>;
+}
+
+export interface GrokProcessingDefinition {
+  grok: Omit<GrokProcessorConfig, 'if'>;
+}
+
+export interface ProcessingDefinition {
+  condition: Condition;
+  config: DissectProcessingDefinition | GrokProcessingDefinition;
+}
```

Comment:
Why don't we use directly the [ProcessorDefinition](https://github.com/elastic/kibana/pull/206930/files#diff-5d9af36d0ce221d05560b84dbd859aac80ae3bd2bec4fdf7f0c2a87e8d8423bcR72) type, which would fit well with the form state and remove a lot of typing to adapt to the previous state shape?

Reply:
As commented above in this PR, I'll take over refactoring this part for scalability.